### DISCUSSION TITLE
Can I use elytron DB-based security with reactive clients?

### DISCUSSION THREAD

**Discussion by cyChop** [quarkusio/quarkus#28929] [Q&A] [answered]
Hi,

I'm starting a new app with Quarkus and I'm learning the reactive way. I need a lightweight authentication and chose elytron-security-jdbc. This seems to be specifically designed for the JDBC extensions, though, not the reactive ones, as it tells me it doesn't find the JDBC driver (unless I change the reactive extension to the JDBC one).

Hence my question: is there a way to use elytron's database-backed security with reactive clients?

I had a quick look at the sources of quarkus-elytron-security-jdbc, and I'm guessing the only source that would need an adaptation would be the io.quarkus.elytron.security.jdbc.JdbcRecorder. Is there a known way to do it, maybe even locally in my own code, so that I can test it out?

[a comment by an automated account is not shown]

**Comment by cyChop** [comment]
Thinking back on my question, I'm wondering if maybe I could write an adapter for components that require the JDBC component. I'll probably have the same issue with other components (I'm typically setting up Liquibase).

Or should I just declare both the jdbc and reactive extensions, with config for both? This feels dirty.

**Comment by sberyozkin** [comment, marked as answer]
Hi @cyChop I believe the reactive equivalents for both of JDBC and JPA security extensions have to be created, the issue for JPA is here: #23553, have a look please if it can be of interest
